src/mem3/src/mem3_util.erl

```diff
@@ -34,6 +35,27 @@ hash(Item) when is_binary(Item) ->
 hash(Item) ->
     erlang:crc32(term_to_binary(Item)).
 
+docid_hash(DocId) when is_binary(DocId) ->
+    docid_hash(DocId, []).
+
+docid_hash(<<"_design/", _/binary>> = DocId, _Options) ->
+    erlang:crc32(DocId); % design docs are never placed by partition
```

Comment:
Assuming the motivation to not use the `hash/1` function is not wanting to call `term_to_binary` on the DocId, I think it would be wise to avoid redefining the use of `erlang:crc32/1` and we should instead call out to a default hash function that is used ubiquitously. Perhaps something like:

```erlang
hash_raw(Item) ->
  erlang:crc32(Item).

hash(Item) ->
  hash_raw(term_to_binary(Item)).

%% ...

%% example usage
docid_hash(DocId, ...) ->
  hash_raw(DocId).
```